controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -221,52 +224,53 @@ func (r *StorageClassRequestReconciler) initPhase(storageProfile *v1.StorageProf
 
 	// check request status already contains the name of the resource. if not, add it.
 	if r.StorageClassRequest.Spec.Type == "blockpool" {
+		// initialize in-memory structs
+		r.cephRadosNamespace = &rookCephv1.CephBlockPoolRadosNamespace{}
+		r.cephRadosNamespace.Namespace = r.OperatorNamespace
 		r.cephBlockPool = &rookCephv1.CephBlockPool{}
+		r.cephBlockPool.Name = profileName
 		r.cephBlockPool.Namespace = r.OperatorNamespace
```

Comment:
There is no real need to have a block pool object on the reconciler. We are not loading it, creating it, or updating it.
In all cases, the name is derived from the profile name on the input (StoargeClassRequest), or if not provided from from the default ODF entities. 
So we don't even need to hold the name on the reconciler object.  
```suggestion
```